common/network_manager.go

```diff
@@ -89,6 +94,7 @@ type NetworkManager struct {
 // NewNetworkManager creates a new network manager.
 func NewNetworkManager(
 	ctx context.Context, customMetrics *k6ext.CustomMetrics, s session, fm *FrameManager, parent *NetworkManager,
+	mi metricInterceptor,
```

Comment:
I'd reformat and lay out these inputs over separate lines as they got long.

Reply:
This is a nit pick but done in 4d9fa0c5c49fabf128adf3721e64f2988b11cef5

Reply:
Yep, it's a small detail. Nothing important. Feel free to skip some of my suggestions. I've already approved the PR. Just noting what I see.